core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -195,4 +195,84 @@ class ReplicaManagerTest {
       metrics.close()
     }
   }
+  
+  @Test
+  def testFetchBeyondHighWatermarkNotAllowedForConsumer() {
```

Comment:
Could we clear the ReplicaManagerTest class a bit, by putting common objects into variables of the class like props / config / zkClient / time / jTime / metrics / etc. And we can probably even centralizing replica manager object as well.